### PR TITLE
Feature/project creation

### LINKED ISSUE #11: Project creation
As a user i need to be able to create projects with name, estitame sprints number, sprints duration

### PR DESCRIPTION
**Pull request by TitusCln**

### CODE REVIEW

**Comment by oscarSantoyo** [comment]
@TitusCln Is this PR ready to review?

File: contracts/src/main/java/com/sbs/contracts/clients/UserStoryProxy.java [hunk 1]
```diff
@@ -3,11 +3,7 @@
 import com.sbs.contracts.dto.TaskDTO;
 import com.sbs.contracts.dto.UserStoryDTO;
 import org.springframework.cloud.openfeign.FeignClient;
-import org.springframework.stereotype.Service;
-import org.springframework.web.bind.annotation.PathVariable;
-import org.springframework.web.bind.annotation.RequestBody;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.*;
```
Comment (oscarSantoyo) [inline contracts/src/main/java/com/sbs/contracts/clients/UserStoryProxy.java]: Please deactivate the folding imports in your IDE https://stackoverflow.com/questions/3348816/intellij-never-use-wildcard-imports

- [x] Use single imports

File: contracts/src/main/java/com/sbs/contracts/clients/UserStoryProxy.java [hunk 2]
```diff
@@ -35,8 +31,11 @@
...
 
   @RequestMapping(method = RequestMethod.PUT, value = prefix + "/{userStoryId}/tasks/{taskId}")
-  public TaskDTO updateTask(@PathVariable(value = "userStoryId")Long userStoryId, @PathVariable(value = "taskId") Long taskId, @RequestBody TaskDTO TaskDTO);
+  public TaskDTO updateTask(@PathVariable(value = "userStoryId") Long userStoryId, @PathVariable(value = "taskId") Long taskId, @RequestBody TaskDTO TaskDTO);
 
   @RequestMapping(method = RequestMethod.DELETE, value = prefix + "/{userStoryId}/tasks/{taskId}")
   public void deleteTask(@PathVariable(value = "userStoryId") Long userStoryId, @PathVariable(value = "taskId") Long taskId);
+
+  @RequestMapping(method = RequestMethod.GET, value = prefix, params = {"ids"})
+  public Iterable<UserStoryDTO> getProjectUserStories(@RequestParam(value = "ids") Long[] ids);
```
Comment (oscarSantoyo) [inline contracts/src/main/java/com/sbs/contracts/clients/UserStoryProxy.java]: I think a better name for this method is `getUserStoriesByIds`

File: projects/src/main/java/com/sbs/projects/models/ProjectSprint.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
+package com.sbs.projects.models;
+
+import com.sbs.contracts.dto.ProjectSprintDTO;
+
+import javax.persistence.*;
```
Comment (oscarSantoyo) [inline projects/src/main/java/com/sbs/projects/models/ProjectSprint.java]: - [x] Use unfold imports.

File: userStories/src/main/java/com/sbs/userStories/UserStoryController.java [hunk 4]
```diff
@@ -22,6 +24,11 @@
     return userStoryService.getAllUserStories();
   }
 
+  @GetMapping(value = "/userstories", produces = MediaType.APPLICATION_JSON_VALUE, params = {"ids"})
+  public Iterable<UserStoryDTO> getUsersStoriesByIds(@RequestParam Long[] ids) {
```
Comment (oscarSantoyo) [inline userStories/src/main/java/com/sbs/userStories/UserStoryController.java]: This naming is correct, proxy had a different one.

File: userStories/src/main/java/com/sbs/userStories/services/impl/UserStoryServiceImpl.java [hunk 5]
```diff
@@ -54,17 +54,24 @@ public UserStoryDTO updateUserStory(UserStoryDTO updatedUserStory, Long id) {
...
 
-//    private Stream<S> getOptionalSteam(Collection<T> iterable){
+    @Override
+    public Iterable<UserStoryDTO> getByIds(Iterable<Long> ids) {
+        return StreamSupport.stream(userStoryRepository.findAllById(ids).spliterator(), true)
+                .map(UserStory::toDTO)
+                .collect(Collectors.toList());
+    }
+
+    //    private Stream<S> getOptionalSteam(Collection<T> iterable){
```
Comment (oscarSantoyo) [inline userStories/src/main/java/com/sbs/userStories/services/impl/UserStoryServiceImpl.java]: Remove commented code.

**Review by oscarSantoyo — CHANGES_REQUESTED** [review]
Done for now, please address my comments.

File: userStories/src/main/java/com/sbs/userStories/services/impl/UserStoryServiceImpl.java (hunk 5 above)
Comment (TitusCln) [reply userStories/src/main/java/com/sbs/userStories/services/impl/UserStoryServiceImpl.java]: this commented code it was for you to check if it was possible to create de common generic method utility for this code repetitive implementation @oscarSantoyo 

`Optional.ofNullable(updatedUserStory.getLabels())
              .orElse(new HashSet<>())
              .parallelStream()`

**Review by oscarSantoyo — APPROVED** [review]
Good job :medal_military:

**Pull request merged**